### ISSUE TITLE
the renter is idle , but tracking active pieces: 8

### ISSUE THREAD

**Issue by Fornax96** [NebulousLabs/Sia#1591]
Here's the stdout / stderr of siad:

```
./siad -M gctwr
Loading...
(0/5) Loading siad...
(1/5) Loading gateway...
(2/5) Loading consensus...
(3/5) Loading transaction pool...
(4/5) Loading wallet...
(5/5) Loading renter...
Finished loading in 7.458458434 seconds
Rescanning consensus set...
Scanned to height 92927...
Done!
goroutine 484 [running]:
runtime/debug.Stack(0x44cc47, 0x0, 0xc4426fbbe0)
        /usr/lib/go-1.7/src/runtime/debug/stack.go:24 +0x79
runtime/debug.PrintStack()
        /usr/lib/go-1.7/src/runtime/debug/stack.go:16 +0x22
github.com/NebulousLabs/Sia/build.Critical(0xc4426fbd30, 0x2, 0x2)
        /home/wim/go/src/github.com/NebulousLabs/Sia/build/critical.go:15 +0xa7
github.com/NebulousLabs/Sia/persist.(*Logger).Critical(0xc42088a140, 0xc4426fbd30, 0x2, 0x2)
        /home/wim/go/src/github.com/NebulousLabs/Sia/persist/log.go:37 +0xd0
github.com/NebulousLabs/Sia/modules/renter.(*Renter).managedDownloadIteration(0xc420a2c000, 0xc4426fbed0)
        /home/wim/go/src/github.com/NebulousLabs/Sia/modules/renter/download.go:312 +0x5d1
github.com/NebulousLabs/Sia/modules/renter.(*Renter).threadedDownloadLoop(0xc420a2c000)
        /home/wim/go/src/github.com/NebulousLabs/Sia/modules/renter/download.go:576 +0x37b
created by github.com/NebulousLabs/Sia/modules/renter.newRenter
        /home/wim/go/src/github.com/NebulousLabs/Sia/modules/renter/renter.go:208 +0x3b3
Critical error: ERROR: the renter is idle , but tracking active pieces: 1
Please submit a bug report here: https://github.com/NebulousLabs/Sia/issues
goroutine 484 [running]:
runtime/debug.Stack(0x44cc47, 0x0, 0xc4426fbbe0)
        /usr/lib/go-1.7/src/runtime/debug/stack.go:24 +0x79
runtime/debug.PrintStack()
        /usr/lib/go-1.7/src/runtime/debug/stack.go:16 +0x22
github.com/NebulousLabs/Sia/build.Critical(0xc4426fbd30, 0x2, 0x2)
        /home/wim/go/src/github.com/NebulousLabs/Sia/build/critical.go:15 +0xa7
github.com/NebulousLabs/Sia/persist.(*Logger).Critical(0xc42088a140, 0xc4426fbd30, 0x2, 0x2)
        /home/wim/go/src/github.com/NebulousLabs/Sia/persist/log.go:37 +0xd0
github.com/NebulousLabs/Sia/modules/renter.(*Renter).managedDownloadIteration(0xc420a2c000, 0xc4426fbed0)
        /home/wim/go/src/github.com/NebulousLabs/Sia/modules/renter/download.go:312 +0x5d1
github.com/NebulousLabs/Sia/modules/renter.(*Renter).threadedDownloadLoop(0xc420a2c000)
        /home/wim/go/src/github.com/NebulousLabs/Sia/modules/renter/download.go:576 +0x37b
created by github.com/NebulousLabs/Sia/modules/renter.newRenter
        /home/wim/go/src/github.com/NebulousLabs/Sia/modules/renter/renter.go:208 +0x3b3
Critical error: ERROR: the renter is idle , but tracking active pieces: 8
Please submit a bug report here: https://github.com/NebulousLabs/Sia/issues
```

Here's another message from renter.log: `2017/02/24 10:59:16.602570 download.go:312: CRITICAL: ERROR: the renter is idle , but tracking active pieces: 8`. The other log files in the renter dir are empty.

OS: Ubuntu Server Edition 16.10
Compiled siad with golang 1.7.1
Siad version 1.1.1-RC3

**Comment by DavidVorick** [comment]
I'm hoping that this is related to the race condition I just fixed, but I'm going to leave this open because I'm not entirely confident that the race condition was the cause.

**Comment by Fornax96** [comment]
I did some more testing. 

This message is printed when a download fails with the message `Could not download file: download failed: insufficient hosts to recover file`. The `tracking active pieces` number also increases by ~10 every time I try a download.

I'm still on the latest master branch.

**Comment by DavidVorick** [comment]
Good to know. I guess the bug is still there, will have to look into it soon. Working on some significant host bugs first though.

**Comment by Fornax96** [comment]
Right now I'm downloading a 20 GB file which I couldn't download yesterday. The only difference is that I restarted the client overnight. 

It seems like one failed download will also make following downloads fail, as I haven't been able to download anything after that message appeared. 

Maybe connections to the host are not being closed properly when a download fails? It seems like it could be something like that. Then the channel stays open and the host cannot be used for other downloads which means all following file downloads cannot reach enough hosts to retrieve the file. 

Anyway, it seems like me this is quite an urgent bug because not being able to download files will definitely harm the user experience. After typing this the 20 GB file I started downloading is still chugging along fine at an average speed of 50 Mbps, so that's going well so far.

**Comment by DavidVorick** [comment]
This is second in my queue to work on, first being catastrophic host failures that are causing hosts to lose all their contracts. (Windows only).

After that I'll be doing some more in-depth testing and verification of the renter code. I'm guessing at least a handful of bugs will show up.

Eta is a few weeks though

**Comment by vevarm** [comment]
I get this in 1.3.0-rc2, think its related to download, (but confusing when logs dont have local timestamps). 
2017/06/27 18:10:47.893362 download.go:368: CRITICAL: ERROR: the renter is idle , but tracking active pieces: 30

**Comment by vevarm** [comment]
Seems this happens only when download is 100%.

**Comment by lukechampine** [comment]
should be fixed in 1.3.0 :)
